src/main/java/com/uber/jenkins/phabricator/PhabricatorNotifier.java

```diff
@@ -231,6 +240,23 @@ public final boolean perform(final AbstractBuild<?, ?> build, final Launcher lau
         return true;
     }
 
+    protected UberallsClient getUberallsClient(Logger logger, String gitUrl, String branch) {
```

Comment:
Not too sure how this is used. But why don't you set uberallsClient and return the instance if it was null to begin with. This way, if this is called multiple times, we return the cached instance.